### PR TITLE
UID2-1485 update keycloak templates

### PR DESCRIPTION
**Pull request by jingyi-gao-ttd**
### Keycloak Deployment Required?
Yes

### Keycloak Config Updates
- [ ]  in Realm settings, switch `User Profile Enabled` to off 

### Changes
1. Updated email copy and stylings

- Reset Password
Screenshot 2023-07-24 at 12 15 09 pm

- Verify Email
Screenshot 2023-07-24 at 12 17 37 pm

- Accept Invite

Screenshot 2023-07-24 at 12 20 58 pm

2. Updated keycloak login theme
- If the email is not verified
image
- Forgot Password
image
image

### CODE REVIEW

File: keycloak/themes/uid2-theme/email/messages/messages_en.properties [hunk 1]
```diff
@@ -1,7 +1,8 @@
...
-inviteBody=<p>Hey {0}, You&rsquo;ve been added to Unified ID Portal.. Please use the link below to create a password for your account.{1}</p>
\ No newline at end of file
+inviteBodyHtml=<div style="text-align: center"> <p style="text-align: left"> Hey {0},<br /><br /> You've been added to the UID2 Portal.<br /><br /> Please use the button below to create a password for your account. </p> </div> <div style="margin: 67px 0"> <a href="{1}" style=" border-radius: 5px; font-weight: 600; font-size: 18px; padding: 10px 19px; margin: 0 auto; background-color: #cdf200; color: #030a40; width: auto; text-align: center; text-decoration: none; " >Accept Invitation</a > </div>
```
Comment (jingyi-gao-ttd) [inline keycloak/themes/uid2-theme/email/messages/messages_en.properties:7]: We cannot get the name of person who send the invite, so I removed the `by {someone}`

File: keycloak/themes/uid2-theme/login/login-verify-email.ftl [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+                                ${kcSanitize(messagesPerField.getFirstError('username','password'))?no_esc}
+                        </span>
+                    </#if>
+
+                </div>
+
+                <div class="${properties.kcFormGroupClass!} ${properties.kcFormSettingClass!}">
+                  <div class="${properties.kcFormOptionsWrapperClass!}">
+                      <#if realm.resetPasswordAllowed>
+                          <span><a tabindex="5" href="${url.loginResetCredentialsUrl}">${msg("doForgotPassword")}</a></span>
```
Comment (jingyi-gao-ttd) [inline keycloak/themes/uid2-theme/login/login-verify-email.ftl:44]: this loginResetCredentialsUrl is not given on verify email page. The user will need a few more clicks to proceed. Should we remove this link or disable it?

File: keycloak/themes/uid2-theme/login/login-verify-email.ftl [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        </#if>
+        </div>
+
+    </div>
+    <#elseif section = "info" >
+        <#if realm.password && realm.registrationAllowed && !registrationDisabled??>
+            <div id="kc-registration-container">
+                <div id="kc-registration">
+                    <span>${msg("noAccount")} <a tabindex="6"
+                                                 href="${url.registrationUrl}">${msg("doRegister")}</a></span>
```
Comment (jingyi-gao-ttd) [inline keycloak/themes/uid2-theme/login/login-verify-email.ftl:63]: Same as above, the register button will not work

File: keycloak/themes/uid2-theme/login/messages/messages_en.properties [hunk 4]
```diff
@@ -1,16 +1,18 @@
...
 infoTitle=Your Information
 passwordTitle=Create a Password
 doRegisterButton=Continue
 errorPatternNoMatch=We’re sorry, but we only accept sign-ups from company email addresses. Please use your official company email to register and try again.
-forgotPasswordInfo=Enter your email address below and we will send a link to reset your password.
-doSendLink=Send Link
+forgotPasswordInfo=Enter your email address and we’ll send a link to reset your password.
+doSendLink=Request Password Reset
 emailForgotTitle=Forgot Password
 updatePasswordTitle=Reset Password
```
Comment (jingyi-gao-ttd) [inline keycloak/themes/uid2-theme/login/messages/messages_en.properties:15]: We can not differentiate if this user goes to this page by invite or forgot password. i saw on figma it says 'Create password' for new users, should we change this for a generic one that works for both?

**Comment by lionell-pack-ttd** [comment]
See my Slack - the copy doesn't make sense now without changing the overall flow in a way I don't think we can do (for security reasons among others), so we're going to have to push back on some of these changes.

**Comment by alex-yau-ttd** [comment]
I see the copy for Accept invite in the PR description is missing an apostrophe - is this screenshot still up-to-date or does this need fixing? 
image

I can see inviteBodyHtml has 'You've' while inviteBody has 'You&rsquo;ve' - does inviteBodyHtml need to be changed then?

**Comment by jingyi-gao-ttd** [comment]
@alex-yau-ttd  Good catch, thanks! i've updated both
image
image

File: docker-compose.yml [hunk 5]
```diff
@@ -26,7 +26,7 @@ services:
       KC_DB_USERNAME: sa
       KC_DB_PASSWORD: D3velopmentP0
       KEYCLOAK_ADMIN: admin
-      KEYCLOAK_ADMIN_PASSWORD: D3velopmentP0
+      KEYCLOAK_ADMIN_PASSWORD: g
```
Comment (alex-yau-ttd) [inline docker-compose.yml:29]: we just want to make this easier for us to type the password?

**Comment by alex-yau-ttd** [comment]
Few nits on the log in page, if it isn't too hard to fix:
image

- "Log In" and "Email" are a bit close, could we please increase the gap here?
- Log In button is wider than the design, though this doesn't really bother me. 
- The design also doesn't have a question mark at the end of "Forgot Password", but I think it makes sense to keep it.
- "Don't have an account?" text is bold in the design

Design, for convenience: 
image

File: keycloak/themes/uid2-theme/login/login-verify-email.ftl (hunk 2 above)
Comment (alex-yau-ttd) [reply keycloak/themes/uid2-theme/login/login-verify-email.ftl:44]: Sorry I'm not quite understanding this - is this a link that doesn't work?

File: docker-compose.yml (hunk 5 above)
Comment (jingyi-gao-ttd) [reply docker-compose.yml:29]: oh! I think I accidentally make this change

**Comment by jingyi-gao-ttd** [comment]
@alex-yau-ttd  I've removed the login-verify-email template for now, as I remember we are not going to build the fake login page for verification messages any more, I will create a separate PR for that once we got the new design

- gap between login and email has been increased.
- Size of the buttons has been fixed.
- It's a bit hard to change "Don't have an account?" and "Request Account " separately. So I set both to 500

File: keycloak/themes/uid2-theme/login/login-verify-email.ftl (hunk 2 above)
Comment (jingyi-gao-ttd) [reply keycloak/themes/uid2-theme/login/login-verify-email.ftl]: I've now removed this page and will create a new ticket to address this once Kristen comes up a new design for this

File: keycloak/themes/uid2-theme/login/resources/css/login.css [hunk 6]
```diff
@@ -76,20 +79,24 @@ body {
...
   font-size: 18px;
-  padding: 15px 120px;
+  padding: 15px 11px;
   margin: 0 auto;
 }
 
 .pf-c-button.pf-m-primary {
   background-color: var(--theme-button);
   color: var(--theme-button-border);
+  min-width: 225px;
```
Comment (alex-yau-ttd) [inline keycloak/themes/uid2-theme/login/resources/css/login.css:96]: Per meeting with Kristen this morning, I think we also want to reduce the height of these buttons. The design uses 45px

Comment (jingyi-gao-ttd) [reply keycloak/themes/uid2-theme/login/resources/css/login.css:96]: done

**Comment by alex-yau-ttd** [comment]
This screen is a bit odd to me - I don't think the info banner should show up here, since the email has already been sent and there is also the button at the bottom to re-send the email.
image
What do you think?

**Comment by jingyi-gao-ttd** [comment]
I remove the copy change on this page, but as the other page wants the message to appear on top, it will look like this now. I will write a message and ask for Kimberley and Kristen's input 
Screenshot 2023-08-03 at 1 59 51 pm

File: keycloak/realm/realm-export.json [hunk 7]
```diff
@@ -2336,7 +2336,7 @@
     "actionTokenGeneratedByUserLifespan-execute-actions": "",
     "actionTokenGeneratedByUserLifespan-verify-email": "",
     "clientOfflineSessionIdleTimeout": "0",
-    "userProfileEnabled": "true",
+    "userProfileEnabled": "false",
```
Comment (jingyi-gao-ttd) [inline keycloak/realm/realm-export.json:2339]: userProfile is managed in self-serve-portal, so we could switch this to off, to make sure it is using default register.ftl template

**Comment by alex-yau-ttd** [comment]
> I remove the copy change on this page, but as the other page wants the message to appear on top, it will look like this now. I will write a message and ask for Kimberley and Kristen's input Screenshot 2023-08-03 at 1 59 51 pm

that seems reasonable to me

**Review by alex-yau-ttd — APPROVED** [review]

**Pull request merged**